src/vm/methodtablebuilder.cpp

```diff
@@ -2900,6 +2900,26 @@ MethodTableBuilder::EnumerateClassMethods()
             }
         }
 
+        // Some interface checks.
+        if (fIsClassInterface && (CLRConfig::GetConfigValue(CLRConfig::UNSUPPORTED_TypeLoader_DefaultInterfaces) == 0))
```

Comment:
These `CLRConfig::GetConfigValue` checks are relatively slow. You should move it to right before throwing exception; or cache it in `static ConfigDWORD`.